.github/workflows/ci.yml

```
@@ -110,15 +110,41 @@ jobs:
       - name: Check
         run: cargo check --no-default-features -p petgraph --target wasm32v1-none --features graphmap,serde-1,stable_graph,matrix_graph,generate,unstable
 
-  miri:
-    name: Unsoundness check
-    if: github.event_name != 'merge_group'
+  fast-miri:
+    # This job runs on pull_request without the `S-run-thorough-ci-tests` label.
+    name: Unsoundness check fast
+    if: ${{ github.event_name != 'merge_group' && github.event_name == 'pull_request' && !contains(github.event.pull_request.labels.*.name, 'S-run-thorough-ci-tests') }}
```

Can be removed, unless you wanted to keep it explicitly. (same with the other `if`)
```suggestion
    if: ${{ github.event_name == 'pull_request' && !contains(github.event.pull_request.labels.*.name, 'S-run-thorough-ci-tests') }}
```